core/node/api_server/src/web3/namespaces/eth.rs

```diff
@@ -689,6 +689,8 @@ impl EthNamespace {
         effective_pubdata_price_history.reverse();
 
         let oldest_block = newest_l2_block.0 + 1 - base_fee_per_gas.len() as u32;
+        // `base_fee_per_gas` for next L2 block cannot be calculated, appending last fee as a placeholder.
+        base_fee_per_gas.push(*base_fee_per_gas.last().unwrap());
```

Comment:
Nice catch! However, length of `gas_used_ratio `, `reward` and `blob_gas_used_ratio` should left untouched as it depends on `base_fee_per_gas.len()`.
Separately, what about length of `l2_pubdata_price`? I think we shouldn't change it, i.e. don't include "next" value because it's zksync extension and we don't need to follow any eth spec here. But I'm curious what do you think about it

Reply:
also, please add an assert for each array len in the test